src/language-mode.coffee

```diff
@@ -246,23 +246,30 @@ class LanguageMode
     iterator.next()
     scopeDescriptor = new ScopeDescriptor(scopes: iterator.getScopes())
 
+    increaseNextIndentRegex = @increaseNextIndentRegexForScopeDescriptor(scopeDescriptor)
+    decreaseNextIndentRegex = @decreaseNextIndentRegexForScopeDescriptor(scopeDescriptor)
+    decreaseIndentRegex = @decreaseIndentRegexForScopeDescriptor(scopeDescriptor)
+
     currentIndentLevel = @editor.indentationForBufferRow(bufferRow)
-    return currentIndentLevel unless increaseIndentRegex = @increaseIndentRegexForScopeDescriptor(scopeDescriptor)
+    return currentIndentLevel unless increaseNextIndentRegex
 
-    if options?.skipBlankLines ? true
+    bufferLine = @buffer.lineForRow(bufferRow)
+    if options?.skipBlankLines ? true or (decreaseIndentRegex?.testSync(bufferLine) and not @editor.isBufferRowCommented(bufferRow))
```

Comment:
Pff... This one doesn't come easy right :disappointed:

Having yet another look...